src/components/Messaging/MessageBody.js

```diff
@@ -0,0 +1,294 @@
+        <Typography variant={'h5'} className={surveyContent}>
+          {message.body}
+        </Typography>
+        {message.survey.questions.map((question, i) => (
+          <div key={i} className={surveyContent}>
+            <Typography variant={'h6'}>
+              Question {i + 1}:{' '}
+              <Typography variant={'body1'}>{question.prompt}</Typography>
+            </Typography>
+            <Typography variant={'h6'}>
+              Choices:
+              <ol type="A">
+                {question.choices.map((choice, j) => (
+                  <li key={j + 1}>{choice}</li>
+                ))}
+              </ol>
+            </Typography>
+          </div>
+        ))}
+      </Grid>
+    </div>
+  );
+};
+
+export const RecievedMessage = ({ message }) => {
+  const {
+    messageRow,
+    messageTimeStampRight,
+    messageContent,
+    recievedMessage,
+  } = useStyles();
+  return (
+    <>
+      <div className={messageRow}>
+        {/* <Avatar alt={''} src={message.authorId ? '' : ''}></Avatar> */}
+        <div className={recievedMessage}>
+          <div>
+            <Typography className={messageContent}>{message.body}</Typography>
+          </div>
+        </div>
+        <Grid item className={messageTimeStampRight}>
+          <Typography>{message.composed_at.slice(0, 10)}</Typography>
+        </Grid>
+      </div>
+    </>
+  );
+};
+
+export const SentMessage = ({ message }) => {
+  const {
+    messageRowRight,
+    messageContent,
+    messageTimeStampLeft,
+    sentMessage,
+  } = useStyles();
+
+  return (
+    <>
+      {message.body ? (
+        <Grid className={messageRowRight} container>
+          <Grid item className={messageTimeStampLeft}>
+            <Typography>{message.composed_at.slice(0, 10)}</Typography>
+          </Grid>
+          <Grid item className={sentMessage}>
+            <Typography className={messageContent}>{message.body}</Typography>
+          </Grid>
+        </Grid>
+      ) : (
+        <div></div>
+      )}
+    </>
+  );
+};
+
+const SenderInformation = ({ messageRecipient, id }) => {
+  const { senderInfo, senderItem, avatar } = useStyles();
+
+  return (
+    <Grid container className={senderInfo}>
+      <Grid item className={senderItem}>
+        <Avatar src="" className={avatar}></Avatar>
+      </Grid>
+      <Grid item className={senderItem}>
+        <Typography variant="h5">{messageRecipient}</Typography>
+        <Typography align="left" color="primary" variant="h5">
+          ID:{id}
+        </Typography>
+      </Grid>
+    </Grid>
+  );
+};
+
+const MessageBody = ({ messages, messageRecipient }) => {
+  const { paper, messagesBody, textInput } = useStyles();
+  const { user, postMessageSend } = useContext(MessagingContext);
+  const [messageContent, setMessageContent] = useState('');
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+
+    const messagePayload = {
+      parent_message_id: messages[messages.length - 1].id
```

Comment:
Could `messages[messages.length - 1]` could be stored in a variable for multiple use?